frontend/coprs_frontend/coprs/models.py

```diff
@@ -2090,30 +2090,41 @@ def result_dir_url(self):
         return urljoin(app.config["BACKEND_BASE_URL"], os.path.join(
             "results", self.build.copr_dir.full_name, self.name, self.result_dir, ""))
 
-    def _compressed_log_variant(self, basename, states_raw_log):
+    def log_variant(self, basename, states_raw_log, compressed):
+        """
+        Return the full URL to the log file with the given basename.
+
+        Args:
+            basename: the basename of the log file (e.g. "builder-live.log")
+            states_raw_log: list of states in which the log file is not compressed
+                (e.g. ["running", "starting"])
+            compressed: whether the log file is compressed (.gz suffix)
+
+        Returns:
+            the full URL to the log file or None if the log file does not exist
+        """
```

Comment:
:shrug: this rewrite is weird, can we have a separate method for the new link?

Reply:
Or IMHO even better, can we improve log detective to handle the compressed logs?